### ISSUE TITLE
TTF Not Drawing Correctly

### ISSUE THREAD

**Issue by HeinousGames** [libgdx/libgdx#5400]
#### Issue details
My TTF is being drawn incorrectly on Android devices using the latest snapshot version. Desktop seems to be fine. My personal game project loads otf files fine, but seems to skip over the ttfs. I have included a link to a sample project below that only contains one affected ttf file.

#### Reproduction steps/code
Below is a link to a sample project. Run it on an Android and you will see that the font displays as [empty squares.](https://github.com/HeinousGames/LibGDXFontIssue/raw/master/Screenshot_20181013-133908.png) Change libgdx version to 1.9.8 to see [how it should normally look.](https://github.com/HeinousGames/LibGDXFontIssue/raw/master/Screenshot_20181013-134249.png) The outcome is the same using both the FreeTypeFont generator or loading through an asset manager.
Here is an [open repo with a modified setup project](https://github.com/HeinousGames/LibGDXFontIssue) to show the results.

#### Version of LibGDX and/or relevant dependencies
1.9.9-SNAPSHOT
MacOS High Sierra v.10.13.6 running Android Studio 3.1.4
Reproducible on Samsung Galaxy S8 running Android 8.0.0 and Android OnePlus2 running Android 6.0.1.

#### Please select the affected platforms
- [X] Android
- [ ] iOS (robovm)
- [ ] iOS (MOE)
- [ ] HTML/GWT
- [ ] Windows
- [ ] Linux
- [ ] MacOS

**Comment by NathanSweet** [comment]
@PokeMMO Any chance this is related to #5247? @HeinousGames since you can repro, does it help to revert the changes from that commit?

**Comment by PokeMMO** [comment]
He doesn't have an aaptOptions defined, so it should be using the legacy copy method.

**Comment by Tom-Ski** [comment]
I'd say this is more likely to be due to https://github.com/libgdx/libgdx/commit/c942c61d5f8e8cf90e91ce3d1414d61cf9b4845e and the IDE is just caching wrong snapshot for desktop builds. Running from command line should give a better idea of the cause.

**Comment by vbousquet** [comment]
One related thing that could be improved simultaneously is that the missing glyph is always added to the font texture as long as it is defined in the font file. For most (game) use case, you will not use it in the end. It just consumes useless space in the texture. 

Therefore, it would seem more logical to only include it if requested by the user. In [generatedData](https://github.com/libgdx/libgdx/blob/master/extensions/gdx-freetype/src/com/badlogic/gdx/graphics/g2d/freetype/FreeTypeFontGenerator.java#L413-L418), something like this ;
```java
        if (parameter.characters.contains("\0")) {
            Glyph missingGlyph = createGlyph('\0', data, parameter, stroker, baseLine, packer);
            if (missingGlyph != null && missingGlyph.width != 0 && missingGlyph.height != 0) {
                data.setGlyph('\0', missingGlyph);
                data.missingGlyph = missingGlyph;
                if (incremental) data.glyphs.add(missingGlyph);
            }
        }
```

Or, if we don't want to make an API change, keep the 'include .notdef glyph by default' and add a flag to disable this behavior.

**Comment by HeinousGames** [comment]
> I'd say this is more likely to be due to [c942c61](https://github.com/libgdx/libgdx/commit/c942c61d5f8e8cf90e91ce3d1414d61cf9b4845e) and the IDE is just caching wrong snapshot for desktop builds. Running from command line should give a better idea of the cause.

I had tried clearing/invalidating my caches a few times and didn't seem to fix the issue before. Also, the desktop version never had the issue, are you implying that the Android build relies on the desktop build somehow? I believe my issue ended up coming from the change for FontAwesome a few days ago, which was also related to the commit you suggested. Funny how one change can make a chain lol. I have invalidated my caches again today and the issue seems to be gone. I'm assuming it's related to Nathan's changes above, but not sure exactly how it fixes the problem. Nonetheless, thank you all for the quick turnaround!
